homeassistant/components/izone/climate.py

```diff
@@ -314,12 +326,38 @@ def current_temperature(self) -> Optional[float]:
             return self._controller.temp_supply
         return self._controller.temp_return
 
+    @property
+    def control_zone_name(self):
+        """Return the zone that currently controls the AC unit."""
+        if not self._supported_features & SUPPORT_TARGET_TEMPERATURE:
+            zone_ctrl = self._controller.zone_ctrl
+            zone = next(
+                (z for z in self.zones.values() if z._zone.index == zone_ctrl), None
+            )
+            if zone is None:
+                return None
+            return zone.name
+        return None
+
+    @property
+    def control_zone_setpoint(self) -> Optional[float]:
+        """Return the temperature setpoint of the zone that currently controls the AC unit."""
+        if not self._supported_features & SUPPORT_TARGET_TEMPERATURE:
+            zone_ctrl = self._controller.zone_ctrl
+            zone = next(
+                (z for z in self.zones.values() if z._zone.index == zone_ctrl), None
+            )
+            if zone is None:
+                return None
+            return zone._zone.temp_setpoint
```

Comment:
Make the `_zone` attribute non private on the zone entity, since we need to access it from outside the zone entity.

Reply:
Better yet: Add a read-only property ```zone_index``` to the ```ZoneDevice``` in ```controller.py``` and use that.

Reply:
And what about the `temp_setpoint` attribute that we need to access here? Should we wrap all zone attributes in properties? What's the point with that?

Reply:
I don't know a lot about python; the code is currently working (deployed it to my home instance), and a quick search indicates python, as a fairly loose language, doesn't have strict visibility, but does do name mangling if you use a double underscore.

So, I guess the single underscore is simply meant to indicate privacy, which is kind of what I had assumed, but that since they were both in the same file that access each other's internals was okay (kind of like the "friend" visibility in some languages).

From the design of the original code it looks like the ZoneDevice (and ControllerDevice) are meant to hide/encapsulate the Zone/Controller from the imported library.

So I'm going back to the original style and going to put back in a property zone_index that wraps _zone.index.

For the reference to _zone.temp_setpoint, it actually already is wrapped in a property, so I will just use that.

i.e. all Zone attributes were already wrapped... the only one not wrapped is zone_index, and that is only because I just changed it.

Reply:
Every other Controller and Zone attribute, and private value, is/was already wrapped, e.g. available, supply_temperature, min_temp, max_temp, zone: current-temperature, target_temperature, etc. I think a lot of these are wrapped to conform with the climate device model.

It was using zone._zone.temp_setpoint in the controller instead of zone.target_temperature. So, yes, all of them are already wrapped.

This also means the implementation of Zone, in the separate interface library, is fully encapsulated (in case it is replaced/changed). i.e. we don't expose _zone.

Reply:
Most of those are part of the Entity interface and follows that in terms of properties. We don't need to add properties if we just need to access an instance attribute. My suggestion is to make the `_zone` attribute non private.

Reply:
@MartinHjelmare I prefer the way @sgryphon  had it, for the reasons they discussed, keeping the library wrapped and not exposing internals (`_zone` is a friend). Second best would be a `zone_index` property. Exposing the Zone object doesn't make a lot of sense to me.

Reply:
Ok. But I still don't understand the point. It's apparently not very internal since we need to access all these attributes of the zone from outside. These are only details in the integration anyway and nothing other than the integration will access any of this.

Reply:
All other functions of Zone (and Controller) were already wrapped in ZoneDevice and ControllerDevice, e.g. I should not have used z._zone.set_temp, but should have just used the existing wrapper z.target_temperature. All these wrappers are needed to conform to the device interface.

Rather than "access all these attributes of zone", the only one attribute (not already mapped) was the index, to match it to the control zone index. To access this one property, I don't think exposing the whole Zone object makes sense.

But as it is all internal, I don't really care, and will write it whichever way will get approved.

@Swamp-Ig, @MartinHjelmare do either of you have the ability to approve, and what changes would you like to see done so that I can get the approval.

Reply:
I'm code owner, but I don't contribute a lot to HA anymore.  I don't know who technically has the greater approval powers here 😀 I don't really mind what happens.

Reply:
I'm fine with the existing approach for now.

But it is generally not good to mix the device api into the Entity api. That's the main problem here, and a consequence of that is also these properties that wrap the zone device api. It would be better if the entities just consumed the device api and the controller didn't access the zone entities at all from outside. That's our standard. If something needs to tell an entity to do something from outside we use our dispatch helper for this.